lib/app.rb

```diff
@@ -10,145 +11,137 @@ class App
   def initialize(output_stream, input_stream)
     @output_stream = output_stream
     @input_stream = input_stream
+    @projects = []
```

Comment:
This is great! Making sure the data is there as an array from the beginning solves so many possible issues.